js/models/userMd.js

```diff
@@ -92,6 +93,84 @@ module.exports = Backbone.Model.extend({
       return modGuid;
     });
 
+    // If blocked_guids has no legitmate guids, it returns an array with an empty string.
+    // Let's remedy that.
+    if (response.blocked_guids.length && response.blocked_guids[0] === "") {
+      response.blocked_guids = [];
+    }
+
     return response;
-  }
+  },
+
+  toJSON: function() {
+    var attributes = __.clone(this.attributes);
+
+    attributes.blocked = attributes.blocked_guids || [];
+
+    return attributes;
+  },
+
+  _blockUnblockUser: function(block, guid) {
+    var self = this,
+        blockedGuids,
+        index,
+        request;
+
+    blockedGuids = this.get('blocked_guids') || [];
+    index = blockedGuids.indexOf(guid);
+
+    if (block && index === -1) {
+      blockedGuids.push(guid);
+    } else if (index !== -1) {
+      blockedGuids.splice(index, 1);
+    }
+
+    this.set('blocked_guids', blockedGuids);
+
+    request = saveToAPI(null, this.toJSON(), this.get('serverUrl') + 'settings', null, '', { blocked: blockedGuids })
+      .done(function(data) {
+        var blocked,
+            i;
+
+        if (data.success) {
+          window.obEventBus.trigger(block ? 'blockedUser' : 'unblockedUser', { 'guid': guid });        
+        } else {
+          // on fail, let's revert the optomistic model updates
+          blocked = self.get('blocked_guids') || [];
+          i = blocked.indexOf(guid);
+
+          if (block && blocked.length && i !== -1) {
+            // a block attempt failed
+            blocked.splice(index, 1);
+          } else if (i === -1) {
+            // an unblock attempt failed
+            blocked.push(guid);
+          }
+          
+          self.set('blocked_guids', blocked);
+        }
+      }).fail(function() {
+        throw new Error('There was an error blocking user ' + guid);
+      });
+
+    window.obEventBus.trigger(block ? 'blockingUser' : 'unblockingUser', { guid: guid, request: request });
+
+    return request;
+  },
+
+  blockUser: function(guid) {
```

Comment:
I've put the functionality to block/unblock a user onto the userModel, which seems (at least to me) like an appropriate centralized place for it. At some point, we should also do the same with the follow/unfollow functionality. Right now, it lives in the Home View, which is not a good place for it if other views wanted to make use of it.